src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -292,6 +292,10 @@ public final void removeInterestConnect(SelectionKey key) {
      * @param key the key.
      */
     public final void addInterestRead(SelectionKey key) {
+        if (LOGGER.isLoggable(Level.FINEST)) {
```

Comment:
>  I suppose you added that because of the TODO below...

yes - easier to do it now and know the performance hit - also prevents someone forgetting to do it later.